src/discretization/regular.jl

```diff
@@ -57,7 +57,8 @@ function discretize(sphere::Sphere{3,T},
   # add north and south poles
   c = center(sphere)
   r = radius(sphere)
-  e⃗ = Vec{3,T}(0, 0, 1)
+  V = T <: AbstractFloat ? T : Float64
+  e⃗ = Vec{3,V}(0, 0, 1)
```

Comment:
This feels like a hack. Can you please explain why we need to hard-code Float64 here? You raised this Unitful issue before and we need to take it seriously.

Reply:
OK.

I admit it looks a little unnatural. But this line of code was inspired by your code. You have wrote it in `src/sampling/regular.jl`. I didn't think much about it as I was writing it down.

Now that you brought this up, after thinking about it, I think it would be more appropriate to write

```julia
V = typeof(one(T))
```

Since `push!(points, c + r*e⃗)` follows immediately below, `e⃗` should be a dimensionless number. The `one` function is used to obtain the dimensionless unit quantity corresponding to a variable.

Reply:
That is a good point. I wonder if we can write down a general recipe for dealing with unitful quantities in these types of calculations. Can you propose a fix using `oneunit` or `one`, whichever is appropriate?

More generally, say we have coordinates x = (x1, x2, x3). These coordinates are written with respect to a basis and an origin point x = x1*E1 + x2*E2 + x3*E3 + O. Should these coordinates have units? Can we say that unitful quantities form a field? When we think more generally about vector spaces over a field, we assume specific properties for the scalars x1, x2, x3.

Reply:
The Unitful developers said [`one` is the multiplicative identity, i.e., `x * one(x) == x`. There is also `oneunit(3u"mm") == 1u"mm"`.](https://github.com/PainterQubits/Unitful.jl/issues/508#issuecomment-994529657)

Thus, we know. `one` is multiplicative identity, could act as a factor. `oneunit` is one with the same unit. With the question x' = Ex + O, x and O should have the same unit like meter. E as a factor should not have any unit but have the same integer or float expression. Which means `typeof(one(E)) == typeof(one(x))`.

What we tend to confuse is whether this is a coefficient or a real physical quantity. If it is a coefficient, i.e. a scaling factor, direction vector, etc., it should not carry units. Whereas if it is a real physical quantity, e.g. position (coordinates), displacement (difference of coordinates) should have units.

Reply:
That is an interesting summary, thanks for sharing. So the rule of thumb is to always use units with coordinates and avoid units with scaling factors.